---
title: "test files"
author: "Jixing"
date: "`r Sys.Date()`"
output: 
  html_document:
    fig_height: 3
    fig_width: 5
  pdf_document:
    fig_height: 3
    fig_width: 5
  word_document:
    fig_height: 3
    fig_width: 5
---

```{r, setup, include=FALSE}
require(mosaic)   # Load additional packages here 
knitr::opts_chunk$set(
  tidy=FALSE,     # display code as typed
  size="small")   # slightly smaller font for code
library(tidyverse)
library(ggpubr)
here <- here::here
select <- dplyr::select
```

## Flowchart
![](/Users/zero/Desktop/RNASeq_calling_variants/07_images/create_testfiles_flowchart.png)

## get bed file for TP53:7157 
  1. input: gene entrezID
  2. ouput: bed, gtf, fasta files
```{r}
# get a transcript database, which stores exon, trancript, and gene information
library(GenomicFeatures)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene

# extracting information from txdb
g <- genes(txdb) # GRanges, just start to end, no exon/intron information

# subset Grange
TP53  <- g[g$gene_id  %in% c("7157")]

```

# bed file
```{r}
rtracklayer::export.bed(TP53, "../02_annotation_file/TP53.bed")
```


# fasta file
```{r}
library(BSgenome.Hsapiens.UCSC.hg19)
fasta <- getSeq(Hsapiens, TP53)
library(Biostrings)
writeXStringSet(fasta, file = "../02_annotation_file/TP53.fa", format = "fasta")
```


# gtf file
```{bash}
grep -E  TP53\" genes.gtf > TP53.gtf
```


## use bed file to subset bam 
```{bash}
samtools view -L your.bed -b your.bam > subset.bam
```


## bam to fastq
```{bash}
bedtools bamtofastq -i subset.bam \
                    -fq subset_1.fq \
                    -fq2 subset_2.fq
```


